src/plugins/intel_cpu/src/nodes/multiclass_nms.cpp

```diff
@@ -6,97 +6,252 @@
+        return boxes + batch_idx * num_boxes_ * 4;
+    }
+
+    const float* scores(const int batch_idx, const int class_idx, const float* scores, const int*) const override {
+        return scores + batch_idx * num_classes_ * num_boxes_ + class_idx * num_boxes_;
+    }
+
+    int num_input_boxes(const int batch_idx, const int* roisnum) const override {
+        return num_boxes_;
+    }
+
+    void locate(const Box& box, const int* roisnum, int& input_box_index, int& input_box_offset) override {
+        const int index = box.batch_idx * num_boxes_ + box.box_idx;
+        input_box_index = index;
+        input_box_offset = index * 4;
+    }
+};
+
+//
+// boxes:   [num_classes, num_boxes, 4] // where `num_boxes` is the sum of boxes from all batches
+// scores:  [num_classes, num_boxes]    // where `num_boxes` is the sum of boxes from all batches
+// roisnum: [num_batches]               // The sum of all elements is `num_boxes`
+//
+class MultiClassNms::InputsFormatSecond : public MultiClassNms::Inputs {
+public:
+    Format format() const override { return Format::second; }
+
+    void prepareParams(const Node& node, const std::string& err_str) override {
+        const auto& boxes_dims = node.getParentEdgeAt(NMS_BOXES)->getMemory().getStaticDims();
+        const auto& scores_dims = node.getParentEdgeAt(NMS_SCORES)->getMemory().getStaticDims();
+        if (boxes_dims[0] != scores_dims[0] || boxes_dims[1] != scores_dims[1])
+            IE_THROW() << err_str << "has incompatible 'boxes' and 'scores' shape " << PartialShape(boxes_dims) << " v.s. " << PartialShape(scores_dims);
+        const auto& roisnum_dims = node.getParentEdgeAt(NMS_ROISNUM)->getMemory().getStaticDims();
+        if (roisnum_dims.size() != 1)
+            IE_THROW() << err_str << "has unsupported 'roisnum' input rank: " << roisnum_dims.size();
+        num_batches_ = roisnum_dims[0];
+        num_boxes_ = boxes_dims[1];
+        num_classes_ = scores_dims[0];
+    }
+
+    const float* boxes(const int batch_idx, const int class_idx, const float* boxes, const int* roisnum) const override {
+        const int batch_offset = std::accumulate(roisnum, roisnum + batch_idx, int{});
+        return boxes + (class_idx * num_boxes_ + batch_offset) * 4;
+    }
+
+    const float* scores(const int batch_idx, const int class_idx, const float* scores, const int* roisnum) const override {
+        const int batch_offset = std::accumulate(roisnum, roisnum + batch_idx, int{});
+        return scores + class_idx * num_boxes_ + batch_offset;
+    }
+
+    int num_input_boxes(const int batch_idx, const int* roisnum) const override {
+        return roisnum[batch_idx];
+    }
+
+    void locate(const Box& box, const int* roisnum, int& input_box_index, int& input_box_offset) override {
+        const int batch_offset = std::accumulate(roisnum, roisnum + box.batch_idx, int{});
+        input_box_index = (batch_offset + box.box_idx) * num_classes_ + box.class_idx; // TODO: bug in reference impl?
```

Comment:
Corresponding code in reference implementation https://github.com/openvinotoolkit/openvino/blob/master/src/core/reference/src/runtime/reference/multiclass_nms.cpp#L532

Reply:
This expression should calculate an index of a box in `boxes` input tensor. The shape of this tensor for this input format is `[num_classes, num_boxes, 4]`. Taking this shape into account, the expression `(batch_offset + box.box_idx) * num_classes_ + box.class_idx;` calculates something I can't understand what. Also, we cant multiply the result of this expression by 4 to obtain the offset of box coordinates. The correct expression has to be `(box.class_idx * num_boxes_ + batch_offset + box.box_idx)`. That is why reference implementation looks suspicious to me.
The fix to the reference implementation could be as the following:
```
@@ -529,7 +529,7 @@ void multiclass_nms(const float* boxes_data,
             for (int64_t i = 0; i < box_info.batch_index; i++) {
                 offset += roisnum_data[i];
             }
-            selected_index = {(offset + box_info.index), box_info.class_index, num_classes};
+            selected_index = {box_info.class_index, (offset + box_info.index), num_boxes};
         }
         SelectedOutput selected_score{static_cast<float>(box_info.class_index),
                                       box_info.score,
```

Reply:
@ceciliapeng2011 
Tried to to fix this bug in CPU plugin and  paddlepaddle frontend tests failed.